src/storage/hummock_sdk/src/key.rs

```diff
@@ -150,6 +152,49 @@ fn next_key_no_alloc(key: &[u8]) -> Option<(&[u8], u8)> {
 
 // End Copyright 2016 TiKV Project Authors. Licensed under Apache-2.0.
 
+/// Get the end bound of the given `prefix` when transforming it to a key range.
+fn end_bound_of_prefix(prefix: &[u8]) -> Bound<Vec<u8>> {
+    let next_key = next_key(prefix);
+    if next_key.is_empty() {
+        Unbounded
+    } else {
+        Excluded(next_key)
+    }
+}
+
+/// Transform the given `prefix` to a key range.
+pub fn range_of_prefix(prefix: &[u8]) -> impl RangeBounds<Vec<u8>> {
+    if prefix.is_empty() {
+        (Unbounded, Unbounded)
+    } else {
+        (Included(prefix.to_vec()), end_bound_of_prefix(prefix))
+    }
+}
+
+/// Prepend the `prefix` to the given key `range`.
+pub fn prefixed_range<B: AsRef<[u8]>>(
```

Comment:
What is it for? It seems error-prone to use. 🥸

Reply:
And it does not avoid misusing next_key

Reply:
Suggest not to add it unless used somewhere

Reply:
My motivation is just to simplify the `CellBasedTable::serialize_pk_bound`. Is there anything I'm missing?

Reply:
Personally, I think this change is ok :)

Reply:
For `serialize_pk_bound`, included end bound should be excluded next key.
e.g., pk is [a,b,c], if we want to scan a=1, b<=2, the range (a,b)<=(1,2) is incorrect……

Reply:
That's why I say it's error prone. So I suggest to do the simplification completely instead of only adding this function.

Reply:
It's keyspace prefix and is for things like `Keyspace::iter_with_range`, not for `CellBasedTable::serialize_pk_bound` :)